presto-test-jdbc-compatibility/src/test/java/io/prestosql/TestJdbcCompatibility.java

```diff
@@ -0,0 +1,479 @@
+            checkDescribeTimestampType(query, "array(%s)", precision, false);
+        }
+    }
+
+    @Test
+    public void testSelectParametricTimestampWithTimeZoneInArray()
+    {
+        if (hasBrokenParametricTimestampWithTimeZoneSupport()) {
+            throw new SkipException("This version raports PARAMETRIC_DATETIME client capability but TIMESTAMP WITH TIME ZONE is not supported");
+        }
+
+        String timestamp = "1984-08-24 23:55:23.383345789012";
+
+        for (int precision = 0; precision <= TIMESTAMP_MAX_PRECISION; precision++) {
+            String query = format("SELECT ARRAY[cast(TIMESTAMP '%s %s' as TIMESTAMP(%d) WITH TIME ZONE)]", timestamp, TESTED_TZ, precision);
+            String expectedTimestamp = roundedTimestamp(timestamp, precision, TIMESTAMP_MAX_PRECISION, true);
+
+            checkRepresentation(query, expectedTimestamp + " " + TESTED_TZ, TestJdbcCompatibility::getSingleElementFromArray);
+            checkDescribeTimestampType(query, "array(%s)", precision, true);
+        }
+    }
+
+    @Test
+    public void testSelectParametricTimestampInRow()
+    {
+        String timestamp = "2014-08-24 23:55:23.383345789012";
+
+        for (int precision = 0; precision <= TIMESTAMP_MAX_PRECISION; precision++) {
+            String query = format("SELECT CAST(ROW(TIMESTAMP '%s') AS ROW(timestamp TIMESTAMP(%d)))", timestamp, precision);
+            String expectedTimestamp = roundedTimestamp(timestamp, precision, TIMESTAMP_MAX_PRECISION, false);
+
+            checkRepresentation(query, expectedTimestamp, TestJdbcCompatibility::getSingleElementFromRow);
+            checkDescribeTimestampType(query, "row(timestamp %s)", precision, false);
+        }
+    }
+
+    @Test
+    public void testSelectParametricTimestampWithTimeZoneInRow()
+    {
+        if (hasBrokenParametricTimestampWithTimeZoneSupport()) {
+            throw new SkipException("This version raports PARAMETRIC_DATETIME client capability but TIMESTAMP WITH TIME ZONE is not supported");
+        }
+
+        String timestamp = "2004-08-24 23:55:23.383345789012";
+
+        for (int precision = 0; precision <= TIMESTAMP_MAX_PRECISION; precision++) {
+            String query = format("SELECT CAST(ROW(TIMESTAMP '%s %s') AS ROW(timestamp TIMESTAMP(%d) WITH TIME ZONE))", timestamp, TESTED_TZ, precision);
+            String expectedTimestamp = roundedTimestamp(timestamp, precision, TIMESTAMP_MAX_PRECISION, true);
+
+            checkRepresentation(query, expectedTimestamp + " " + TESTED_TZ, TestJdbcCompatibility::getSingleElementFromRow);
+
+            checkDescribeTimestampType(query, "row(timestamp %s)", precision, true);
+        }
+    }
+
+    @Test
+    public void testSelectMongoObjectId()
+    {
+        String query = "SELECT ObjectId('55b151633864d6438c61a9ce') AS objectId";
+        checkRepresentation(query, new byte[]{85, -79, 81, 99, 56, 100, -42, 67, -116, 97, -87, -50}, ResultSet::getObject);
```

Comment:
test `getString` as well

also, what does `java.sql.ResultSetMetaData#getColumnType` report for object id?

Reply:
getString won't work - it returns Object hash, not an actual representation (new byte[]{})

Reply:
i meant `ResultSet::getString`